pkg/cache/internal/informers_map.go

```diff
@@ -162,9 +164,22 @@ func (ip *specificInformersMap) Get(gvk schema.GroupVersionKind, obj runtime.Obj
 	}
 
 	if started && !i.Informer.HasSynced() {
+		syncReturn := make(chan bool)
```

Comment:
make this buffered so that we don't leak the goroutine when we time out

```suggestion
		syncReturn := make(chan bool, 1 /* don't leak goroutines on timeout */)
```